v2/cmd/asoctl/internal/importing/importable_arm_resource.go

```diff
@@ -326,6 +326,14 @@ func (*importableARMResource) classifyError(err error) (string, bool) {
 			// If we're not allowed to look, we can still import the rest
 			return "access forbidden", true
 		}
+
+		if responseError.StatusCode == http.StatusBadRequest &&
+			strings.Contains(responseError.Error(), "RequestUrlInvalid") {
+			// We constructed an invalid URL
+			// (Seems that some extension resources aren't permitted in all locations)
```

Comment:
What exactly does "not permitted in all locations" mean? Are you referring to location here as an Azure region? Or do you mean "URL location"?

Maybe an example of when this might happen?

Or alternatively, is there a way we can know that a URL is/isn't bad and just not construct it in the first place? Rather than inferring from response that we must have generated something bad but it's OK? For example if this method is used to classify errors for both standard and extension resources, presumably this `RequestUrlInvalid` is not expected for a standard resource, but we'd ignore it anyway?

Reply:
Got a couple of errors where the constructed URL for listing extension resources underneath another resource returned `RequestUrlInvalid`. I've managed to tweak the logic to mostly avoid doing this (by avoiding nesting of extension resources) but it still came up in a couple cases.